scripts/frontmatter/get-frontmatter.js

```diff
@@ -0,0 +1,93 @@
+const frontmatter = require('@github-docs/frontmatter');
+const { TYPES } = require('../constants');
+
+const GATSBY_CONTENT_TYPES = {
+  [TYPES.BASIC_PAGE]: 'page',
+  [TYPES.API_DOC]: 'apiDoc',
+  [TYPES.RELEASE_NOTE]: 'releaseNote',
+  [TYPES.RELEASE_NOTE_PLATFORM]: 'releaseNotePlatform',
+  [TYPES.TROUBLESHOOTING]: 'troubleshootingDoc',
+  [TYPES.WHATS_NEW]: 'nr1Announcement',
+};
+
+const GATSBY_TEMPLATE = {
+  [TYPES.BASIC_PAGE]: 'basicDoc',
+  [TYPES.API_DOC]: 'basicDoc',
+  [TYPES.RELEASE_NOTE]: 'basicDoc',
+  [TYPES.RELEASE_NOTE_PLATFORM]: 'basicDoc',
+  [TYPES.TROUBLESHOOTING]: 'basicDoc',
+  [TYPES.WHATS_NEW]: 'basicDoc',
+};
+
+const FRONTMATTER_TEMPLATE = {
+  [TYPES.BASIC_PAGE]: 'page',
+  [TYPES.API_DOC]: 'apiDoc',
+  [TYPES.RELEASE_NOTE]: 'releaseNote',
+  [TYPES.RELEASE_NOTE_PLATFORM]: 'releaseNotePlatform',
+  [TYPES.TROUBLESHOOTING]: 'troubleshooting',
+  [TYPES.WHATS_NEW]: 'whatsNew',
+};
+
+const getFrontmatter = (type, doc) => {
+  const defaultFrontmatter = {
+    title: doc.title.replace(':', '-'),
+    contentType: GATSBY_CONTENT_TYPES[type],
+    template: GATSBY_TEMPLATE[type],
+    topics: getTopics(doc),
+  };
+
+  return frontmatter.stringify('', {
+    ...defaultFrontmatter,
+    ...addCustomFrontmatter[FRONTMATTER_TEMPLATE[type]](doc),
```

Comment:
Rather than defining a new object (`FRONTMATTER_TEMPLATE`), we might be able to just use the type as the key for `addCustomeFrontmatter`:

```js
...addCustomFrontmatter[type](doc);
```

In order to do this, you would need to update the object below from:

`apiDoc: (doc) => {` to `[TYPES.API_DOC]: (doc) => {`

The advantage there is that we can remove an extra object and we don't need to rely on another set of keys. It's totally possible that I'm missing something tho!

Reply:
+1 to this suggestion

Reply:
I also like this idea! I will make this change